sprint 14

KonOsipov opened the pull request:
Спринт 14

KonOsipov committed:
sprint 14

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/PlayerActivity.kt:
```diff
@@ -38,10 +56,13 @@ class PlayerActivity : AppCompatActivity() {
         playButton = findViewById(R.id.play_button)
         actualTime = findViewById(R.id.actual_time)
-        actualTime?.text = "00:30"
+        actualTime?.text = "00:00"
```
Лучше не хардкодить этот текст, так как форматирование может легко поменяться и потом придётся по всему проекту искать эти захардкоженные строчки. Можно либо вынести в strings.xml, чтобы это было только в одном месте, либо использовать ту же функцию форматирования через `SimpleDateFormat`, передавая туда значение 0, чтобы поведение было консистентным

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/PlayerActivity.kt:
```diff
@@ -57,5 +78,98 @@ class PlayerActivity : AppCompatActivity() {
+    }
+    override fun onPause() {
+        super.onPause()
+        val secondPlayerStats = playerState
+        if (playerState == STATE_PAUSED) pauseTimer()
+        pausePlayer()
+        playerState = secondPlayerStats
```
А зачем здесь пытаться сохранить `playerState`? Плеер же по факту ставится на паузу, `playerState` тогда начнет некорректное состояние в себе держать

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -262,12 +278,21 @@ class SearchActivity : AppCompatActivity(), TrackHolder.Listener {
-        startActivity(displayIntent)
+        if (clickDebounce()) {
+            val displayIntent = Intent(this, PlayerActivity::class.java)
+            displayIntent.putExtra("track", track)
```
Ключи для передачи данных лучше вынести в константу, т.к. они используются в более чем 1 месте. Ещё один плюс вынесения в константы - в будущем легче будет этот экран переиспользовать

ArturNurtdinov reviewed (commented):
1. Цвет кнопок добавления в избранное и создания плейлиста не соответствуют макетам:
![image](https://github.com/user-attachments/assets/df0eb56c-9df2-443d-9b0b-c72e58f994ad)
Макет:
![image](https://github.com/user-attachments/assets/54b8bff2-a055-468d-8dc2-72dabe0f1874)
В макете используется цвет #1A1B2240 с альфа каналом (RGBA)
2. Размеры иконок в поле ввода не соответствуют макетам:
![image](https://github.com/user-attachments/assets/3ae838be-9e5a-4e0d-8b69-18eecd69c129)
Макет:
![image](https://github.com/user-attachments/assets/8a08ea9d-6d6a-4587-81a7-f4eb0207c145)
В макете иконки 16х16 с внутренними отступами

KonOsipov committed:
sprint 14.1
1."Цвет кнопок добавления в избранное и создания плейлиста нe соответствуют макетам:"
Импортировал напрямую из фигмы через SVG, проверил - цвет в студии #401A1B22,то есть соответствует тому что в Фигме. Вижу что визуально отличается, но у меня реально нет идей как это поправить.
Буду благодарен за помощь.

2."Не работает история поиска:"
Поправил.

3."Размеры иконок в поле ввода не соответствуют макетам"
Поправил.
